src/scripts/validate.sh

```diff
@@ -1,2 +1,2 @@
 #!/bin/bash
-circleci orb validate --skip-update-check "${ORB_PARAM_OUTPUT_DIR}orb.yml"
+circleci orb validate --host "${CIRCLE_API_HOST}" --token "${CIRCLE_TOKEN}" --skip-update-check "${ORB_PARAM_OUTPUT_DIR}orb.yml"
```

Comment:
There is a second issue here, the `lint` job is not designed to require the publishing context or the circleci-token as it is normally not required.

Running against circleci cloud, any token will work because none is required. Using the syntax like the following, we can set the token to the proper value for server customers if they have set it, and a "dummy" value for cloud users.
```
${CIRCLE_TOKEN:-foo}
```

We should also add to the pack job documentation that server customers will need to add their CircleCI token for this job.

Reply:
Unresolving convo to bump the above message

Reply:
Hi @KyleTryon 

thanks so much for the follow-up here 🙇 

I would love to address your above points too.

[the `lint` job](https://github.com/CircleCI-Public/orb-tools-orb/blob/master/src/jobs/lint.yml), as I understand calls on [the _lint.sh_ script here](https://github.com/CircleCI-Public/orb-tools-orb/blob/master/src/scripts/lint.sh).
ultimately, the script uses `yamllint ./src` to lint the YAML files; it does not require a CircleCI CLI nor any token.

I understood you perhaps meant the _validate.sh_ script, which runs `circleci orb validate`.
I have since updated via this commit: https://github.com/CircleCI-Public/orb-tools-orb/pull/161/commits/47542d19027bd1b2068a567ba48bc6a3990fe1be

Specifically, I've:

1. added description to the `pack` job to remind users to set the CIRCLE_TOKEN env var
2. set up a dummy value for CIRCLE_TOKEN in the _validate.sh_ script
3. added conditional step to verify CIRCLE_TOKEN env var is indeed set if self-hosted CircleCI

Reply:
i was thinking to add additional tests under the test-deploy yml but I dont know if we can run tests for expected failures (e.g., testing the verification catches CIRCLE_TOKEN not set when self-hosted).